test/multiverse/suites/hybrid_agent/parsing_helpers.rb

```diff
@@ -123,47 +115,48 @@ def matches_assertion(parameters)
     assert_equal expected, actual, "Expected #{object} to equal #{expected}"
   end
 
-  def verify_agent_output(harvest, output, type)
-    puts "Agent Output: #{type}: #{output[type]}" if ENV['ENABLE_OUTPUT']
+  def verify_agent_output(output)
+    verify_transaction_output(output['transactions'])
+    verify_span_output(output['spans'])
+  end
 
-    if output[type].empty?
-      assert_empty harvest, "Agent output expected no #{type}. Found: #{harvest}"
-    else
-      outputs = prepare_keys(output[type])
-      events = harvest.flatten.select { |a| a.key?('type') }
+  def verify_transaction_output(output)
+    txns = harvest_transaction_events![1].flatten.map { |t| t['name'] }.compact
 
-      events.each_with_index do |event, i|
-        output = outputs[i]
-        if output && event
-          msg = "Agent output for #{type.capitalize} wasn't found in the harvest." \
-            "\nHarvest: #{event}\nAgent output: #{output}"
+    assert_equal output.length, txns.length, 'Wrong number of transactions found'
 
-          if output['parent_name']
-            result = events.find { |e| e['guid'] == event['parentId'] }.dig('name')
+    output.each do |txn|
+      assert_includes(txns, txn['name'], "Transaction name #{txn['name']} missing from output")
+    end
+  end
 
-            assert_equal output['parent_name'], result, msg
-            output.delete('parent_name')
-          end
+  def verify_span_output(output)
+    spans = harvest_span_events![1].flatten.reject { |h| h.empty? }
 
-          assert event >= output, msg
```

Comment:
Previously, the agent output was checked in its entirety to make sure every element matched the cross agent tests. However, that's not the intention of the tests, there may be more data in the harvest. Since that's the case, now we assert on every attribute.